RockPaperScissors/RockPaperScissors.cs

```diff
@@ -6,20 +6,140 @@ class Program
     {
         public static void Main()
         {
-            Console.WriteLine("Enter hand 1:");
-            string hand1 = Console.ReadLine().ToLower();
-            Console.WriteLine("Enter hand 2:");
-            string hand2 = Console.ReadLine().ToLower();
-            Console.WriteLine(CompareHands(hand1, hand2));
+            if (tests())
+            {
+                Console.WriteLine("Tests Passed!");
+            }
+            else
+            {
+                Console.WriteLine("Tests Failed!");
+            }
+            Console.WriteLine("-----------------");
+
+            string hand1 = null;
+
+
+            while (hand1 == null)
+            {
+                try
+                {
+                    playerInput();
+                }
+                catch
+                {
+                    Console.WriteLine("Enter Rock Paper or Scissor. Come on, you know better than that.");
+                }
+                finally
+                {
+                    hand1 = playerInput();
```

Comment:
you calling this in the finally block does not make sense. not sure why it is both here and in the try block.